src/reflect/scala/reflect/runtime/SynchronizedTypes.scala

```diff
@@ -59,7 +58,7 @@ private[reflect] trait SynchronizedTypes extends internal.Types { self: SymbolTa
   private lazy val _undoLog = mkThreadLocalStorage(new UndoLog)
   override def undoLog = _undoLog.get
 
-  private lazy val _intersectionWitness = mkThreadLocalStorage(perRunCaches.newWeakMap[List[Type], sWeakRef[Type]]())
+  private lazy val _intersectionWitness = mkThreadLocalStorage(perRunCaches.newWeakMap[List[Type], WeakReference[Type]]())
```

Comment:
I don't know where this code actually ends up being effectively accessed, but I am a bit worried that there are similar problems of returning `null` where it is not meant

Reply:
They are referenced in Typer.